### DISCUSSION TITLE
Version for Visual Studio Mac

### DISCUSSION THREAD

**Discussion by WaldemarCichon** [metalama/Metalama#102] [Random]
I am using dotnet also to write MacOS and iOS-Apps through Xamarin and now MAUI. 
I wrote a special Proxy Aspect which is working like a charm. But there is no possibility to compile this on the mac, because metalama Nugat is not available for Mac. For the first I do not need the tooling, which is very useful on Windows. But I would need the metalama NuGet available also for the VS-Mac.

**Comment by svick** [comment]
You are right that there is no IDE integration, but otherwise Metalama should work fine on a Mac. If you're getting some error, can you post it here?

**Reply by WaldemarCichon** [reply]
Thanks, I was sure, that I could not find it - but yes, I found it and will try. It is so useful. But please allow a question: is using of parameters in the annotation possible?

**Reply by svick** [reply]
If you mean having parameters in the aspect constructor and then specifying them through the attribute, then yes. [A simple example:](https://try.metalama.net/#N4IgZiBcDaoHZRABQE4HsDmKCGBbAdAMYDOIANCIYgK7ECWcGABALICmALtgDZ7b4AxHLjYB3NCgDW+AILEADm0IdiAbgA6cTYV7FiTAKIAPPPO5tNwTUxtNoAdRR0ObdnuwY2ACnUgAEmzc3GgAhL4AlAC61rYMHEwyACaJXkxxTNhkaXDxAEZM4TE2VnC2ZUwonNQopdhMANRMuRqltgC+mh1acPLUudx0hEw62HpMjs6ubO6eMhwcTrnULkyQTADyAG5sKE6JUxwAFmiJcorKlkVM8k6b2CuV2IlocNwAnkzECwzMIjNsLSuvX6g3GThcbmIHjYc2+SxcXi+TkYTD+UM8hVaxSuZSOdGI+DR0KYAF5UdN0QCrl0gX0BkM0NtdnR9kxEm84HhBgB+DZMvYHY4pTFlErlWwAYRexDQ5nwExcABkGN5yf8Ci1xTYcbZKhxqqURFx8Kg0IQ2GxhZr2p1yCBJIgOCA2pE2kA)

```c#
class Example
{
    [WriteMessage("Hello!")]
    int Add(int a, int b)
    {
        return a + b;
    }
}

public class WriteMessageAttribute : OverrideMethodAspect
{
    private readonly string message;

    public WriteMessageAttribute(string message)
    {
        this.message = message;
    }

    public override dynamic? OverrideMethod()
    {
        Console.WriteLine(message);
        
        return meta.Proceed();
    }
}
```

This generates:

```c#
class Example
{
    [WriteMessage("Hello!")]
    int Add(int a, int b)
    {
        Console.WriteLine("Hello!");
        return a + b;
    }
}
```
